dev-manuals/services/integrate-new-minigame.md

```diff
@@ -85,31 +59,16 @@ This is needed because the configuration ID of tasks must be unique. Add the cas
 
 ```java
 case <YOUR_MINIGAME_NAME>:
-    if (minigameTask.getConfigurationId() == null) {
-        return new MinigameTask(
-            Minigame.<YOUR_MINIGAME_NAME>,
-            minigameTask.getDescription(),
-            null,
-            minigameTask.getIndex()
-        );
-    } else {
-        <YOUR_MINIGAME_NAME>Configuration config = <YOUR_MINIGAME_NAME>Client.getConfiguration(
-            currentAccessToken,
-            minigameTask.getConfigurationId()
-        );
-        config.setId(null);
-        config.getQuestions().forEach(question -> question.setId(null));
-        config = <YOUR_MINIGAME_NAME>Client.postConfiguration(currentAccessToken, config);
-        return new MinigameTask(
-            Minigame.<YOUR_MINIGAME_NAME>,
-            minigameTask.getDescription(),
-            config.getId(),
-            minigameTask.getIndex()
-        );
-    }
+        cloneId = <YOUR_MINIGAME_NAME>Client.postClone(accessToken, minigameTask.getConfigurationId());
+        break;
 ```
 The code snippet above is an example to clone a configuration with questions.
 
+## Integrate cloning into the minigame:
+
+The backend of the minigame has to provide a route to clone a configuration preferably at `/configurations/{id}/clone`.
+This logic should deep copy a configuration for expample by implementing the clone() interface in every object.
```

Comment:
```suggestion
This logic should deep copy a configuration (without the object IDs) for example by implementing the `clone()` method in every object.
Additionally, the minigame should validate that only lecturers can clone a course.
It can do this by requiring a cookie (we recommend the cookie name `access_token` to conform with all other minigames) that is a [JWT token](https://jwt.io/) that, when verified, should have the role `lecturer`.
```